assets/js/modules/analytics-4/components/dashboard/EnhancedMeasurementActivationBanner/index.js

```diff
@@ -41,72 +45,34 @@ import {
 	ACTIVATION_STEP_SETUP,
 	ACTIVATION_STEP_SUCCESS,
 	ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY,
-	ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY,
 } from '../../../constants';
 import { useTooltipState } from '../../../../../components/AdminMenuTooltip/useTooltipState';
 import { useShowTooltip } from '../../../../../components/AdminMenuTooltip/useShowTooltip';
 import { AdminMenuTooltip } from '../../../../../components/AdminMenuTooltip/AdminMenuTooltip';
 import InProgressBanner from './InProgressBanner';
 import SetupBanner from './SetupBanner';
 import SuccessBanner from './SuccessBanner';
-import { MONTH_IN_SECONDS, trackEvent } from '../../../../../util';
-import whenActive from '../../../../../util/when-active';
-import {
-	isValidPropertyID,
-	isValidWebDataStreamID,
-} from '../../../utils/validation';
-import useViewContext from '../../../../../hooks/useViewContext';
-
-function EnhancedMeasurementActivationBanner() {
-	const viewContext = useViewContext();
 
+export default function EnhancedMeasurementActivationBanner( {
+	id,
+	Notification,
+} ) {
 	const [ step, setStep ] = useState( ACTIVATION_STEP_SETUP );
-	const [
-		isEnhancedMeasurementInitiallyDisabled,
-		setIsEnhancedMeasurementInitiallyDisabled,
-	] = useState( undefined );
 	const [ isSaving, setIsSaving ] = useState( false );
 	const [ errorNotice, setErrorNotice ] = useState( null );
 
-	const propertyID = useSelect( ( select ) =>
-		select( MODULES_ANALYTICS_4 ).getPropertyID()
-	);
-
-	const webDataStreamID = useSelect( ( select ) =>
-		select( MODULES_ANALYTICS_4 ).getWebDataStreamID()
-	);
-
+	// See tooltip TODO below being actioned in #10003
 	const isBannerDismissed = useSelect( ( select ) =>
 		select( CORE_USER ).isItemDismissed(
-			ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY
+			'enhanced-measurement-notification'
 		)
 	);
-
 	const isDismissingBanner = useSelect( ( select ) =>
 		select( CORE_USER ).isDismissingItem(
-			ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY
+			'enhanced-measurement-notification'
 		)
 	);
-
-	const hasModuleAccess = useSelect( ( select ) =>
-		select( CORE_MODULES ).hasModuleOwnershipOrAccess( 'analytics-4' )
-	);
-
-	const isEnhancedMeasurementStreamEnabled = useSelect( ( select ) => {
-		if (
-			! isValidPropertyID( propertyID ) ||
-			! isValidWebDataStreamID( webDataStreamID ) ||
-			! hasModuleAccess ||
-			isBannerDismissed
-		) {
-			return undefined;
-		}
-
-		return select( MODULES_ANALYTICS_4 ).isEnhancedMeasurementStreamEnabled(
-			propertyID,
-			webDataStreamID
-		);
-	} );
+	const hideCTABanner = isBannerDismissed || isDismissingBanner;
```

Comment:
Is the banner is dismissed, but `isBannerDismissed` first loads as `undefined` while loading, won't this logic cause a flickering banner?

Eg. the banner will briefly appear because `isBannerDismissed || isDismissingBanner === false` while `isBannerDismissed === undefined`, but then `isBannerDismissed || isDismissingBanner === true` while `isBannerDismissed === true`.

Reply:
So, we've applied this "double protection" hack which comes into play only when we are actually dismissing the banner and not when the Banner loads for the very first time. The notifications API does check for this notification's dismissal state already before loading the banner. If loaded, `isBannerDismissed` will never be "undefined".

This "hack" is just a regular pattern we've applied to several banners that use the AdminMenuTooltip rendering all of which will have this logic removed in #10003.